src/EventGridExtension/EventGridExtensionConfigProvider.cs

```diff
@@ -11,31 +11,40 @@
 using System.Web;
 using Microsoft.Azure.EventGrid;
 using Microsoft.Azure.EventGrid.Models;
+using Microsoft.Azure.WebJobs.Description;
 using Microsoft.Azure.WebJobs.Host.Bindings;
 using Microsoft.Azure.WebJobs.Host.Config;
 using Microsoft.Azure.WebJobs.Host.Executors;
+using Microsoft.Azure.WebJobs.Logging;
 using Microsoft.Extensions.Logging;
 using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
 
 namespace Microsoft.Azure.WebJobs.Extensions.EventGrid
 {
-    public class EventGridExtensionConfig : IExtensionConfigProvider,
+    /// <summary>
+    /// Defines the configuration options for the SendGrid binding.
+    /// </summary>
+    [Extension("EventGrid")]
+    internal class EventGridExtensionConfigProvider : IExtensionConfigProvider,
                        IAsyncConverter<HttpRequestMessage, HttpResponseMessage>
     {
         private ILogger _logger;
+        private readonly ILoggerFactory _loggerFactory;
         private readonly Func<EventGridAttribute, IAsyncCollector<EventGridEvent>> _converter;
 
         // for end to end testing
-        internal EventGridExtensionConfig(Func<EventGridAttribute, IAsyncCollector<EventGridEvent>> converter)
+        internal EventGridExtensionConfigProvider(Func<EventGridAttribute, IAsyncCollector<EventGridEvent>> converter, ILoggerFactory loggerFactory)
         {
             _converter = converter;
+            _loggerFactory = loggerFactory;
         }
 
         // default constructor
-        public EventGridExtensionConfig()
```

Comment:
Decided against cleaning these constructors up to take an `IEventGridClient`? That would make this process cleaner and enhance the tests. If there's a good reason not to do that now, let's at least have an issue open to track.

Reply:
I am going to follow up with another PR for that change. Have updated the rest as per the PR comments.